### PR TITLE
patchkernel: improve manual update

### PR DESCRIPTION
**Pull request by andrea-iob**
After manual modifications to a patch (e.g., addition or deletion of new cells), it is now mandatory to call the "update()" function. This function will take care of updating all patch data structures (adjacencies, ghost exchange information, bounding box, ...).

The same function allows to initialize a partitioned patch: 
 - before adding vertices/cells to the patch, the (empty) patch should be partitioned with the function "PatchKernel::partition(MPI_Comm)";
 - internal and ghost cells should then be added;
 - calling the update function will create the data structures needed for ghost data exchange.  

(Test "test_surfunstructured_parallel_00007" shows how to initialize a partitioned patch.)

This branch include also the changes of pull request #80.

### CODE REVIEW

**Comment by edoardolombardi** [comment]
I pushed two fixing commits:
- the assignement during the partition of the rank owner value not set to -1 in case of distributed mesh;
- in the isDistributed method opposite condition check on the owner value used;

**Comment by andrea-iob** [comment]
I've added a function to initialize the adjacencies: "initializeAdjacencies". 

The function "buildAdjacencies" is now deprecated.

**Comment by edoardolombardi** [comment]
@andrea-iob in the temporary pushed commit d718e68 I added an output to a test useful to check a possible bug in the interfaces update.
The interface ids of some cells seem wrong (unintialized values?) after the second partitioning of the test. 
The same happens when the interfaces are updated after some add/delete operations on cells. In my test the problem was related with ghost cells.

File: src/patchkernel/patch_kernel_parallel.cpp [hunk 1]
```diff
@@ -3460,18 +3470,64 @@ void PatchKernel::unsetGhostCellOwner(int id)
...
 	m_ghostCellOwners.clear();
+
+	setPartitioningInfoDirty(true);
+}
+
+/*!
+	Checks if the information related to the partitioning are dirty.
+
+	\result Returns true if the information related to the partitioning are
+	dirty, false otherwise.
```
Comment (edoardolombardi) [inline src/patchkernel/patch_kernel_parallel.cpp:3555]: Maybe, we can inform that is a collelctive function. The same for other collective get-functions.

File: src/patchkernel/patch_kernel.cpp [hunk 2]
```diff
@@ -4880,6 +4893,38 @@ void PatchKernel::setAdjacenciesBuildStrategy(AdjacenciesBuildStrategy status)
 	m_adjacenciesBuildStrategy = status;
 }
 
+/*!
+	Checks if the adjacencies are dirty.
+
```
Comment (edoardolombardi) [inline src/patchkernel/patch_kernel.cpp:4835]: Here, missing parameter global doc

File: src/patchkernel/patch_kernel.cpp [hunk 3]
```diff
@@ -5237,32 +5268,187 @@ void PatchKernel::setInterfacesBuildStrategy(InterfacesBuildStrategy status)
 	m_interfacesBuildStrategy = status;
 }
 
+/*!
+	Checks if the interfaces are dirty.
+
```
Comment (edoardolombardi) [inline src/patchkernel/patch_kernel.cpp:5197]: Missing documentation of parameter global

**Review by edoardolombardi — APPROVED** [review]

**Pull request merged**